### PR TITLE
DiagnosisKey: Change type long to int for rolling period and rolling start number 

### PR DESCRIPTION
**Pull request by Akshatt**
Solved #233 
- type changed to int 
- rolling start number renamed as rolling start interval number

### CODE REVIEW

File: common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/DiagnosisKey.java [hunk 1]
```diff
@@ -55,11 +55,11 @@
   @Size(min = 16, max = 16, message = "Key data must be a byte array of length 16.")
   private byte[] keyData;
 
-  @ValidRollingStartNumber
-  private long rollingStartNumber;
+  @ValidRollingStartIntervalNumber
+  private int rollingStartIntervalNumber;
 
   @Min(value = 1L, message = "Rolling period must be greater than 0.")
```
Comment (jonaskoepper) [inline common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/DiagnosisKey.java:61]: Also change `value` attribute of annotation to int.

**Comment by jonaskoepper** [comment]
Further places can be cleaned-up, for example:
- https://github.com/corona-warn-app/cwa-server/blob/34646c7db049a5bff031b9692a69d0fcfc6a8777/services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/structure/file/TemporaryExposureKeyExportFile.java
- https://github.com/corona-warn-app/cwa-server/blob/a6199993d3d8b4f035c87b3ebe3c8cf8165414e6/common/persistence/src/test/java/app/coronawarn/server/common/persistence/domain/DiagnosisKeyTest.java#L41

**Comment by Akshatt** [comment]
Can I get a checklist of files where further cleaning up can be done?

**Comment by jonaskoepper** [comment]
I usually look for usages of the property in my IDE (IntelliJ) and search for the property name as string literal also.
IntelliJ has refactoring capability for type changes, too.

**Comment by Akshatt** [comment]
hi @jonaskoepper, I seem to have run into an error while updating DiagnosisKeyBuilder.java : `Cannot resolve symbol protocols`
Here is the screenshot of the error message:
Screenshot (169)

and here is the screenshot of the file in question:
Screenshot (170)

I haven't edited this part of code at all and I can't figure out how the error came to be. Can you shed some light on this?

**Comment by christian-kirschnick** [comment]
@Akshatt Maybe try mvn compile again - looks like you are missing the proto file creation to Java classes.

File: common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/DiagnosisKey.java [hunk 2]
```diff
@@ -105,15 +105,15 @@ public Long getId() {
   /**
    * Returns a number describing when a key starts. It is equal to startTimeOfKeySinceEpochInSecs / (60 * 10).
    */
-  public long getRollingStartNumber() {
-    return rollingStartNumber;
+  public int getRollingStartNumber() {
```
Comment (michael-burwig) [inline common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/DiagnosisKey.java:108]: Please adjust according to variable name change.

File: common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/DiagnosisKeyBuilder.java [hunk 3]
```diff
@@ -59,13 +59,13 @@ public RollingStartNumberBuilder withKeyData(byte[] keyData) {
   }
 
   @Override
-  public RollingPeriodBuilder withRollingStartNumber(long rollingStartNumber) {
+  public RollingPeriodBuilder withRollingStartNumber(int rollingStartNumber) {
```
Comment (michael-burwig) [inline common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/DiagnosisKeyBuilder.java:62]: Please adjust according to variable name change.

File: common/persistence/src/test/java/app/coronawarn/server/common/persistence/domain/DiagnosisKeyTest.java [hunk 4]
```diff
@@ -37,8 +37,8 @@
 class DiagnosisKeyTest {
 
   final static byte[] expKeyData = "testKey111111111".getBytes(Charset.defaultCharset());
-  final static long expRollingStartNumber = 1L;
-  final static long expRollingPeriod = 2L;
+  final static int expRollingStartNumber = 1;
```
Comment (michael-burwig) [inline common/persistence/src/test/java/app/coronawarn/server/common/persistence/domain/DiagnosisKeyTest.java:40]: Please adjust according to variable name change.

**Comment by michael-burwig** [comment]
@Akshatt, thanks for putting the work into this. As @christian-kirschnick pointed out, it looks like you have to run ``mvn clean install`` in either the ``common`` or ``protocol`` module in order to generate the respective Java files from their protobuf specification.

File: common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/DiagnosisKeyBuilder.java [hunk 5]
```diff
@@ -98,8 +95,13 @@ public DiagnosisKey build() {
       submissionTimestamp = Instant.now().getEpochSecond() / 3600L;
     }
 
+<<<<<<< HEAD
```
Comment (michael-burwig) [inline common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/DiagnosisKeyBuilder.java:98]: .

File: common/persistence/src/test/java/app/coronawarn/server/common/persistence/domain/DiagnosisKeyServiceMockedRepositoryTest.java [hunk 6]
```diff
@@ -16,28 +16,29 @@
...
-import java.util.ArrayList;
-import java.util.List;
 import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
 import org.springframework.boot.test.mock.mockito.MockBean;
 import org.springframework.data.domain.Sort;
 import org.springframework.data.domain.Sort.Direction;
 
+import java.nio.charset.Charset;
```
Comment (michael-burwig) [inline common/persistence/src/test/java/app/coronawarn/server/common/persistence/domain/DiagnosisKeyServiceMockedRepositoryTest.java:28]: please, sanitize diff

**Comment by michael-burwig** [comment]
Hi @Akshatt, something must have gone wrong during merge conflict resolution. Coud you please check again and make sure ``mvn install`` runs clean on the root directory.

**Comment by Akshatt** [comment]
hi @michael-burwig, on running `mvn install` I'm getting checkstyle violations in the persistence module, all related to imports.

**Comment by stevesap** [comment]
@Akshatt as you are using InteliJ, you can configure it to use GoogleStyle for the Java CodeStyle and make sure 'Hard wrap at' = 120. Then just use the 'Reformat Code' command on your file (Typically by pressing Ctrl + Alt + l) and that should fix the checkstyle violations.

image

**Comment by Akshatt** [comment]
Hey @stevesap, thanks a lot for that tip! 
All the modules are running except `submission` which gives test failures and the error generated is 
`Caused by: org.hibernate.tool.schema.spi.SchemaManagementException: Schema-validation: wrong column type encountered in column [rolling_period] in table [diagnosis_key]; found [bigint (Types#BIGINT)], but expecting [integer (Types#INTEGER)]` 
This is a database related error, I guess? I don't know how to change that.

**Comment by stevesap** [comment]
@Akshatt you will need to update the database schema to match the new data type from `bigint` to `int`.

You can do so by creating 2 new database migration files named `V2_createTables.sql` under the https://github.com/corona-warn-app/cwa-server/tree/master/common/persistence/src/main/resources/db/migration/h2 and https://github.com/corona-warn-app/cwa-server/tree/master/common/persistence/src/main/resources/db/migration/postgres directories. 

Let me know if you require any additional help.

**Comment by Akshatt** [comment]
> @Akshatt you will need to update the database schema to match the new data type from `bigint` to `int`.
> 
> You can do so by creating 2 new database migration files named `V2_createTables.sql` under the https://github.com/corona-warn-app/cwa-server/tree/master/common/persistence/src/main/resources/db/migration/h2 and https://github.com/corona-warn-app/cwa-server/tree/master/common/persistence/src/main/resources/db/migration/postgres directories.
> 
> Let me know if you require any additional help.

Hi @stevesap,  
Should I replace `V1_createTables.sql` with `V2_createTables.sql` (updated) or keep both of them?
If I keep both of them then I get an error saying table `DIAGNOSIS_KEY` already exists which is due to `V1_createTables.sql`

**Comment by johanneseschrig** [comment]
Hi @Akshatt! You need to keep both files. V2 is a migration from V1, this means you only want a script that runs the delta. You should create a file `V2_changeLongToIntForRollingNumbers.sql` or similar. In V2, you will want to add something like `ALTER TABLE diagnosis_key ALTER COLUMN rolling_period TYPE int;` for the respective columns. Thanks!

**Comment by pithumke** [comment]
Linking #304 as a reminder.

**Comment by Akshatt** [comment]
Hi @johanneseschrig @stevesap @michael-burwig, thanks for all the help. `mvn install` runs clean on the root directory! 
Screenshot (175)

**Comment by michael-burwig** [comment]
Thanks looking fine. I am going to push some minor clean-ups to your fork branch if that's okay.

**Comment by Akshatt** [comment]
> Thanks looking fine. I am going to push some minor clean-ups to your fork branch if that's okay.

Great!

**Pull request merged**